src/components/AddBook.js

```diff
@@ -0,0 +1,33 @@
+import { useDispatch } from 'react-redux';
+import { addBook } from '../Redux/books/bookSlice';
+
+const getData = () => {
+  const titleField = document.querySelector('#title');
+  const authorField = document.querySelector('#author');
+  const titleText = titleField.value;
+  const authorText = authorField.value;
+  titleField.value = '';
+  authorField.value = '';
+  return {
+    title: titleText,
+    author: authorText,
+  };
+};
```

Comment:
Well done overall on the project 💯 but here you are directly interacting with the `DOM` which you shouldn't do in React, instead your component should have a local state for `title` and `author`, which should be updated through the `onChange` event. Consider using the `useState` hook:

```jsx
  const [title, setTitle] = useState("")
  const [author, setAuthor] = useState("")
```

Reply:
I have solved all the issues mentioned as per instructed. Thanks for the kind suggestion